### PR TITLE
Rename NextNetworkUpgrade to Metachain

### PR DESCRIPTION
**Pull request by shohamc1**

### CODE REVIEW

File: test/functional/rpc_blockchain.py [hunk 1]
```diff
@@ -212,7 +212,7 @@ def _test_getblockchaininfo(self):
                     "active": False,
                     "height": 10000000,
                 },
-                "nextnetworkupgrade": {
+                "Metachain": {
```
Comment (prasannavl) [inline test/functional/rpc_blockchain.py]: Case?

File: test/functional/feature_dusd_loans.py [hunk 2]
```diff
@@ -135,10 +135,10 @@ def goto_gce_height(self):
 
     def goto_next_height(self):
         blockHeight = self.nodes[0].getblockcount()
-        if self.nextnetworkupgradeheight > blockHeight:
-            self.nodes[0].generate((self.nextnetworkupgradeheight - blockHeight) + 2)
+        if self.metachainheight > blockHeight:
+            self.nodes[0].generate((self.metachainheight - blockHeight) + 2)
         blockchainInfo = self.nodes[0].getblockchaininfo()
-        assert_equal(blockchainInfo["softforks"]["nextnetworkupgrade"]["active"], True)
+        assert_equal(blockchainInfo["softforks"]["Metachain"]["active"], True)
```
Comment (prasannavl) [inline test/functional/feature_dusd_loans.py:141]: Case again

File: src/rpc/blockchain.cpp [hunk 3]
```diff
@@ -1545,7 +1545,7 @@ UniValue getblockchaininfo(const JSONRPCRequest& request)
     BuriedForkDescPushBack(softforks, "fortcanningepilogue", consensusParams.DF19FortCanningEpilogueHeight);
     BuriedForkDescPushBack(softforks, "grandcentral", consensusParams.DF20GrandCentralHeight);
     BuriedForkDescPushBack(softforks, "grandcentralepilogue", consensusParams.DF21GrandCentralEpilogueHeight);
-    BuriedForkDescPushBack(softforks, "nextnetworkupgrade", consensusParams.DF22NextHeight);
+    BuriedForkDescPushBack(softforks, "Metachain", consensusParams.DF22Metachain);
```
Comment (prasannavl) [inline src/rpc/blockchain.cpp]: Case

File: test/functional/feature_dusd_loans.py (hunk 2 above)
Comment (shohamc1) [reply test/functional/feature_dusd_loans.py:141]: Fixed in 8b51126bc0a87b835a5209b4a197ca65fc61ebc5

File: src/rpc/blockchain.cpp (hunk 3 above)
Comment (shohamc1) [reply src/rpc/blockchain.cpp]: Fixed in 8b51126bc0a87b835a5209b4a197ca65fc61ebc5

File: src/chainparams.cpp [hunk 4]
```diff
@@ -1338,7 +1338,7 @@ void SetupCommonArgActivationParams(Consensus::Params &consensus) {
     UpdateHeightValidation("Fort Canning Epilogue", "-fortcanningepilogueheight", consensus.DF19FortCanningEpilogueHeight);
     UpdateHeightValidation("Grand Central", "-grandcentralheight", consensus.DF20GrandCentralHeight);
     UpdateHeightValidation("Grand Central Epilogue", "-grandcentralepilogueheight", consensus.DF21GrandCentralEpilogueHeight);
-    UpdateHeightValidation("Next Network Upgrade", "-nextnetworkupgradeheight", consensus.DF22NextHeight);
+    UpdateHeightValidation("Next Network Upgrade", "-metachainheight", consensus.DF22Metachain);
```
Comment (prasannavl) [inline src/chainparams.cpp:1341]: Name

Comment (shohamc1) [reply src/chainparams.cpp]: Fixed: f0c313f50bfcc86eaea109bb0f1ec4205f01cb97

File: test/functional/feature_dusd_loans.py (hunk 2 above)
Comment (shohamc1) [reply test/functional/feature_dusd_loans.py]: Fixed: f0c313f50bfcc86eaea109bb0f1ec4205f01cb97

**Pull request merged**